integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, AsyncGenerator
+from httpx import Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from enum import StrEnum
+from port_ocean.context.event import event
+
+class CacheKeys(StrEnum):
+    TEAM = "team"
+    LIST = "list" # project
+    TASK = "task" # issue
+    SPACE = "space"
+    FOLDER = "folder"
+
+class ClickupClient:
+    def __init__(self, clickup_url: str, clickup_apikey: str)-> None:
+        self.clickup_url = clickup_url
+        self.clickup_apikey = clickup_apikey
+        self.api_auth_header = {"Authorization": self.clickup_apikey}
+        self.client = http_async_client
+        self.client.headers.update(self.api_auth_header)
+        self.client.timeout = Timeout(30)
+    
+    @staticmethod
+    def _generate_base_req_params(
+        page: int = 0
+    ) -> dict[str, Any]:
+        return {
+            "page": page,
+        }
+    
+    async def get_teams(self)-> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting teams from Clickup")
+        url = f"{self.clickup_url}/team"
+        team_response = await self.client.get(url)
+        team_response.raise_for_status()
+        team_list = team_response.json()["teams"]
+        yield team_list
+    
+
+
+    async def get_spaces(self)-> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting spaces from Clickup using team_ids from get_teams")
+        async for team_list in self.get_teams():
+            for team_dict in team_list:
+                team_id = team_dict["id"]
+                url = f"{self.clickup_url}/team/{team_id}/space"
+                space_response = await self.client.get(url)
+                space_response.raise_for_status()
+                space_list = space_response.json()["spaces"]
+                # record the present team id to be useful for relation in get_projects() function
+                event.attributes[CacheKeys.TEAM]=team_id
```

Comment:
Good job on caching the response. The whole purpose of caching something is just so we don't have to make another call to the API. At the moment, you are only caching the team ID, and I don't think that is useful information